planetiler-custommap/src/main/java/com/onthegomap/planetiler/custommap/configschema/SchemaConfig.java

```diff
@@ -0,0 +1,55 @@
+package com.onthegomap.planetiler.custommap.configschema;
+
+import java.util.Collection;
+
+/**
+ * An object representation of a vector tile server schema. This object is mapped to a schema YML file using SnakeYAML.
+ */
+public class SchemaConfig {
```

Comment:
You can simplify these data model objects quite a bit by using records:

```java
public record SchemaConfig(
  String schemaName,
  String schemaDescription,
  String attribution,
  Collection<DataSource> sources,
  Collection<FeatureLayer> layers
) {}
```

But, to do that you need to do a 2-step parse using snake yaml to parse the YAML in a way that supports anchors and aliases, then use jackson's object mapper which supports records. For an example see the openmaptiles schema parsing code:

https://github.com/onthegomap/planetiler/blob/6a7c6791416c06fe6bbbcdebaf74cf078e1c2ea4/planetiler-basemap/src/main/java/com/onthegomap/planetiler/basemap/Generate.java#L117-L122

Because the record syntax is so compact, I sometimes like to just include the other data model objects as nested classes so the definition looks similar to example config files, but I don't have a strong preference on that.

Reply:
Resolved in 9e494eb